mitodl/mitxpro
# Data consent agreement models and API functions

#### Pre-Flight checklist


#### What are the relevant tickets?
- Partially addresses #152

#### What's this PR do?
- Create models and some API functions for data consent agreements and users
- Adds `data_consents` to the Basket serializer (list of `DataConsentUser` objects) - will get or create a `DataConsentUser` object for the basket user for any applicable `DataConsentAgreement`s based on the basket's product/coupon.
- Adds `data_consents` patch functionality to the basket view.  The data should be a list of `DataConsentUser` id's, the objects will be assigned a `consent_date` indicating they are signed.

#### How should this be manually tested?
- Tests should pass
- With the fixture `ecommerce.conftest.basket_and_agreement` as a guide, you can manually create a combo of DataConsentAgreement, Product, Course, Coupon, etc objects and apply them to your basket.
- Go to `/api/basket` and verify that there is an entry in `data_consents`.
- Send a patch request to your basket with data `{"data_consents": [<your DataConsentUser id>]}`.  That `DataConsentUser` object should now have a valid `consent_date`
